### PR TITLE
Fix floats in a full-bleed world

### PR DESCRIPTION
**Pull request by jasmussen**
This PR seeks to address #784 by applying an invisible and `calc`ulated outer margin to floats. This margin just happens to match the space surrounding the 700px max-width of the visual editor, making it look right.

I realize the calc looks advanced, but it really simplifies the responsive CSS.

### CODE REVIEW

File: blocks/library/image/style.scss [hunk 1]
```diff
@@ -9,11 +9,19 @@
 	&[data-align="left"] {
 		float: left;
 		margin-right: $block-padding;
+
+		@include break-medium() {
+			margin-left: calc( 50% - #{ $visual-editor-max-width / 2 } );
+		}
```
Comment (youknowriad) [inline blocks/library/image/style.scss]: Aside: Do we expect block authors to implement this CSS styles (which are not that easy) in each floated block

**Review by youknowriad — COMMENTED** [review]
I don't know if it's related but I'm still seeing an issue with floated blocks if they precede an embed block

**Comment by youknowriad** [comment]
Here is the undesired behaviour

screen shot 2017-05-16 at 10 34 09

File: blocks/library/image/style.scss (hunk 1 above)
Comment (jasmussen) [inline blocks/library/image/style.scss]: Good aside. And no, I've been thinking of it this way:

- Full-width can be implemented in many different ways, we've chosen one of many ways
- Theme authors should probably opt-in to the feature, probably something like `add_theme_support( 'full-width-images' );`
- When the theme has opted in, images simply get `align-full-width` as a CSS class. Or something in that vein.

**Comment by jasmussen** [comment]
Thanks for the screenshot, taking a look!

File: blocks/library/image/style.scss (hunk 1 above)
Comment (youknowriad) [inline blocks/library/image/style.scss]: I'm asking about block authors too. I mean, should we move these styles outside the `image` block stylesheet and provide a way to reuse it across different blocks.

Comment (jasmussen) [inline blocks/library/image/style.scss]: Ah yes, good point. Taking a look.

Comment (jasmussen) [inline blocks/library/image/style.scss]: Should we move the visual styles out as part of this PR, or as part of for example #729? Tackling that one seems like a good successor to this one, as video should also have full width as an option.

**Comment by jasmussen** [comment]
Upon discussing this with @mtias I've decided to close this branch, go back to the max-width, then take a pass at using JS for full-width images.

**Pull request closed without merging**